### ISSUE TITLE
Define minimal valuable implementation 

### ISSUE THREAD

**Issue by saschagrunert** [saschagrunert/seccomp-operator#1]
### Target
Create an operator which synchronizes seccomp profiles from config maps to file system locations.

### Use cases and possible issues
TBD

**Comment by pjbgf** [comment]
I was wondering whether is it worth adding this too:

- A user wants to ensure the appropriate ACL permissions are set for all profiles across all nodes.
  - The operator will create directories and profiles with only read & write permissions (`0600`) to the user running the Kubelet.
  - The operator will consolidate ACL permissions to ensure that files inside `seccomp-root/operator` is only read and write by the user running the Kubelet.

**Comment by saschagrunert** [comment]
> I was wondering whether is it worth adding this too:
> 
> * A user wants to ensure the appropriate ACL permissions are set for all profiles across all nodes.
>   
>   * The operator will create directories and profiles with only read & write permissions (i.e. `0600`) to the user running the Kubelet.
>   * The operator will consolidate ACL permissions to ensure that files inside `seccomp-root/operator` have read and write permissions (i.e. `0600`) only to the user running the Kubelet.

Sure, sounds good. :)

**Comment by pjbgf** [comment]
On the "delete" function for MVP we could reconcile only when the entire `ConfigMap` was deleted. It would then check the control plane for containers referring to the profiles that are to be deleted and raise a failure event in case they are still in use. Then the entire physical folder (that represents the given `ConfigMap`) would be deleted.

Later on, once we have a CRD we could add validation to block profiles in use from being deleted in the first place. That actually would actually resolve some of the scenarios of zombie profiles.

**Comment by saschagrunert** [comment]
> On the "delete" function for MVP we could reconcile only when the entire `ConfigMap` was deleted. It would then check the control plane for containers referring to the profiles that are to be deleted and raise a failure event in case they are still in use. Then the entire physical folder (that represents the given `ConfigMap`) would be deleted.
> 
> Later on, once we have a CRD we could add validation to block profiles in use from being deleted in the first place. That actually would actually resolve some of the scenarios of zombie profiles.

Sounds like a plan, we could also rethink that we only allow one profile per config map, but this seems very restrictive, right?

**Comment by pjbgf** [comment]
I would agree. I think the challenge on trying to allow only one profile per config map is that we won't be able to validate that at creation time, which may impact user experience. Feels like one of those decisions to be deferred until we decide to support customs `CRD`'s.

**Comment by saschagrunert** [comment]
@hasheddan @pjbgf I'd prefer that we cut-out the next necessary steps to achieve this goal and put them into separate issues. WDYT?

**Comment by saschagrunert** [comment]
Closing this one, we can handle the open bits in dedicated issues.
